src/main/scala/it/pagopa/interop/authorizationprocess/api/impl/ClientApiServiceImpl.scala

```diff
@@ -684,7 +690,7 @@ final case class ClientApiServiceImpl(
       eService   <- catalogManagementService.getEService(purpose.eserviceId)
       agreements <- agreementManagementService.getAgreements(purpose.eserviceId, purpose.consumerId)
       agreement  <- agreements
-        .filter(_.state != AgreementManagementDependency.AgreementState.PENDING)
+        .filter(agreement => validAgreementsStates.contains(agreement.state))
```

Comment:
Please pay attention here.
I enabled this operation only for Active and Pending agreement states.
Tell me your opinion.
🙇